azext_iot/digitaltwins/providers/resource.py

```diff
@@ -71,6 +72,25 @@ def create(
                 long_running_operation_timeout=timeout,
             )
 
+            def check_state(lro):
+                from time import sleep
+                instance = lro.resource().as_dict()
+                state = instance.get('provisioning_state', None)
+                retries = 0
+                while state == "Provisioning" and retries < MAX_ADT_CREATE_RETRIES:
+                    retries += 1
+                    sleep(1)
+                    lro.update_status()
+                    instance = lro.resource().as_dict()
+                    state = instance.get('provisioning_state', None)
+                if retries == MAX_ADT_CREATE_RETRIES:
+                    logger.warning(
```

Comment:
I would like to know if it is better to raise an error or keep as a warning.

Reply:
The msrest ARM poller has the following provisioning state values

`FINISHED = frozenset(['succeeded', 'canceled', 'failed'])`
`FAILED = frozenset(['canceled', 'failed'])`
`SUCCEEDED = frozenset(['succeeded'])`

If its still provisioning (not a terminal state) at the end of our polling then I think a warning is good. However if a failed provisioning state has been reached and nothing else handles it then an error should be thrown.

Reply:
There isn't any try excepts within the code or poller code (armpoller and the lropoller) that would suggest an error would be caught and not thrown again. 

Is there a way to create a dt instance that would fail but not on the initial call? (So we go into the check_state)

Reply:
I believe it's possible - Essentially since the resource is still `provisioning` after create (and the poller returns to us) - it's definitely plausible that something happens inside that final provisioning step where we'd run a GET and get a failed provisioning state.